velox/duckdb/conversion/DuckWrapper.cpp

```diff
@@ -131,13 +131,34 @@ std::string DuckResult::getName(size_t columnIdx) {
   return queryResult_->names[columnIdx];
 }
 
+bool isNotZeroCopyEligible(const ::duckdb::LogicalType& duckType) {
+  if (duckType.id() == LogicalTypeId::DECIMAL &&
+      (duckType.InternalType() == PhysicalType::INT16 ||
+       duckType.InternalType() == PhysicalType::INT32)) {
+    return true;
+  }
+  if (duckType.id() == LogicalTypeId::HUGEINT ||
+      duckType.id() == LogicalTypeId::TIMESTAMP ||
+      duckType.id() == LogicalTypeId::VARCHAR) {
+    return true;
+  }
+  return false;
+}
+
+inline bool hasValidityVector(::duckdb::Vector& duckVector) {
```

Comment:
The argument can just be the Type?
Do only these types have a validity vector?

Reply:
Renamed as isCreateValidityVector. Not sure why only these cases are considered for copying the validity indices.